### DISCUSSION TITLE
What are all these sanitisers and validators for?

### DISCUSSION THREAD

**Discussion by joulev** [joulev/ezkomment#120] [Questions] [answered]
Just now I checked out at your branch [`be-redesign-and-sanitize-requests`](https://github.com/joulev/ezkomment/tree/be-redesign-and-sanitize-requests) and saw a lot of validators and sanitisers (just as the branch name indicates). So many of them that it'll take anyone quite some time to understand how they integrate together. Why do we need them?

We will probably need them when we implement the public API, but that's a story far far away. Not now.

Let me clear this straight: do not work on sites and pages routes yet. What we are working on is authentication and authentication only. We will work in steps and do not work on step 10 from step 1. That's simply how a project is supposed to progress. I'm waiting for your step 1 to produce something good enough, I don't need your step 10 now.

On the context of authentication, I don't see how we need sanitisers and validators. The APIs are internal and I know how to make valid API calls. We don't need to care about users making erroneous API calls &ndash; since the only supposed user here is me (there might be people spamming the API for fun but if they make invalid calls there is no need to care about them).

Today I spent 2 hours writing [`joulev-backend`](https://github.com/joulev/ezkomment/tree/joulev-backend) which as of [`4d6197e`](https://github.com/joulev/ezkomment/commit/4d6197ece4d41365ea37e21d810e4f07277cec1c) is pretty much everything I need you to do for authentication. 2 hours including learning `firebase-admin/auth` and `firebase-admin/storage`. $31+39+7+19+19+7+12+16=150$ lines of code.  There's a reason why even Vostok teams have to finish authentication by Milestone 1 (29 May) (in that metric we are worse than Vostok teams currently). Stop overcomplicating stuff by custom middlewares and utilities, I don't see how they are necessary at all. We are super super low on time now, 11 days away from the deadline of

* (wip) authentication
* add/get/update/delete sites
* add/get/update/delete pages
* add/get/update/delete comments
* iframe source
* sites having statistics

And don't forget the video and README which I won't do 100% this time.

I'm being annoying &ndash; I know &ndash; but my mental health is taking a hit from this.

**Comment by VietAnh1010** [comment, marked as answer]
Noted, the current method to authenticate is to use JWT.

Just to check again, we will use session cookie to authenticate right? I will implement an endpoint that receive an id token and sent back session cookie in the header.

Are there any other requirements for authentication features?

**Reply by joulev** [reply]
You are free to choose how we authenticate, but given the current time constraint I'd suggest outsourcing all authentication to client Firebase. Only the upload photo step &ndash; which cannot be done in client Firebase unless we set up the bug-prone security rules &ndash; needs to be done serverside.

**Reply by VietAnh1010** [reply]
I used the same strategy for all endpoints: validate the JWT token using FIrebase Admin. 

In client, call `getIdToken()` to get the token, then attach the token into the header of any request. The token will be verified by middlewares, the request is handled only after the validation is complete.

If this strategy is not secure enough, I will implement extra logic.

**Reply by VietAnh1010** [reply]
I completed it, is there any other requirement?

**Reply by joulev** [reply]
I don’t require anything from backend per se, I will blend the frontend to it. Provided the code is good (which I believe is the case). If I require anything from backend it’s speed.

**Reply by VietAnh1010** [reply]
The development speed? Noted. If there are any requirements please tell me right away.

**Reply by joulev** [reply]
Anou, when will the PR land?

**Reply by VietAnh1010** [reply]
Oop sorry, I forgot to open a pull request. Wait a little bit, I will open it.
